tests/model/test_ocrd_file.py

```diff
@@ -1,97 +1,138 @@
-        assert mets.get_physical_pages(for_fileIds=['FOO_1']) == ['p0001']
-        f1.ID = 'BAZ_1'
-        assert mets.get_physical_pages(for_fileIds=['FOO_1']) == [None]
-        assert mets.get_physical_pages(for_fileIds=['BAZ_1']) == ['p0001']
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from tests.base import (
+    main, 
+    create_ocrd_file, 
+    create_ocrd_file_with_defaults
+)
+from ocrd_models import (
+    OcrdMets,
+    OcrdFile,
+)
+
+
+def test_ocrd_file_without_id():
+    with pytest.raises(ValueError) as val_err:
+        create_ocrd_file('FOO')
+
+    assert "set ID" in str(val_err.value)
+
+
+def test_ocrd_file_without_filegrp():
+    with pytest.raises(ValueError) as val_err:
+        create_ocrd_file(None, ID='foo')
+    assert "set fileGrp" in str(val_err.value)
+
+
+def test_set_loctype():
+    f = create_ocrd_file_with_defaults()
+    assert f.loctype == 'OTHER'
+    assert f.otherloctype == 'FILE'
+    f.otherloctype = 'foo'
+    assert f.otherloctype == 'foo'
+    f.loctype = 'URN'
+    assert f.loctype == 'URN'
+    assert f.otherloctype == None
+    f.otherloctype = 'foo'
+    assert f.loctype, 'OTHER'
+
+
+def test_set_url():
+    f = create_ocrd_file_with_defaults()
+    f.url = None
+    f.url = 'http://foo'
+    f.url = 'http://bar'
+    assert f.url == 'http://bar'
+
+
+def test_constructor_url():
+    f = create_ocrd_file_with_defaults(url="foo")
+    assert f.url == 'foo'
+    assert f.local_filename == 'foo'
+
+
+def test_set_id_none():
+    f = create_ocrd_file_with_defaults()
+    f.ID = 'foo12'
+    assert f.ID == 'foo12'
+    f.ID = None
+    assert f.ID == 'foo12'
+
+
+def test_basename():
+    f = create_ocrd_file_with_defaults(local_filename='/tmp/foo/bar/foo.bar')
+    assert f.basename == 'foo.bar'
+
+
+def test_basename_from_url():
+    f = create_ocrd_file_with_defaults(url="http://foo.bar/quux")
+    assert f.basename == 'quux'
+
+
+@pytest.mark.parametrize("local_filename,extension",
+                         [('/tmp/foo/bar/foo.bar', '.bar'),
+                          ('/tmp/foo/bar/foo.tar.gz', '.tar.gz')]
+                         )
+def test_create_ocrd_file_with_defaults_extension(local_filename, extension):
+    """Behavior for ocrd_file_with_defaults
+    """
+
+    f = create_ocrd_file_with_defaults(local_filename=local_filename)
+    assert f.extension == extension
+
+
+@pytest.mark.parametrize("local_filename,wo_extension",
+                         [('/tmp/foo/bar/foo.bar', 'foo'),
+                          ('/tmp/foo/bar/foo.tar.gz', 'foo')]
+                         )
+def test_create_ocrd_file_with_defaults_basename_wo_extension(local_filename, wo_extension):
+    """Behavior for ocrd_file_with_defaults
+    """
+
+    f = create_ocrd_file_with_defaults(local_filename=local_filename)
+    assert f.basename_without_extension == wo_extension
+
+
+@pytest.mark.xfail(reason="not possible anymore as of Fri Sep  3 13:11:00 CEST 2021")
```

Comment:
When do you use `xfail` and when `skip` and when remove the test altogether? I would probably just remove the test, since it's replaced with `test_file_group_wo_parent_new_version`.

Reply:
If it really does serve as documentation, it'll be best to keep it - otherwise drop it. But since it's already part of the repository history (and with the increase of entropy after all), don't be afraid to free yourself  :smile:

Reply:
I'll try to place the col restriction with 150 in the config appropriately

Reply:
Ok I've added to tox.ini, and at least my IDE respects it